data/uhc_pack/functions/pre_generation/stage/generate.mcfunction

```diff
@@ -0,0 +1,19 @@
+scoreboard players set X uhcPG 0
+scoreboard players set Z uhcPG 0
+scoreboard players set Done uhcPG 0
+scoreboard players set Total uhcPG 0
+scoreboard players set Direction uhcPG 1
+
+scoreboard players operation Total uhcPG += UHC uhcBSize
+# This is a little gross, but operation doesn't allow for using literals
+scoreboard players set ChunkSize uhcPG 16
+scoreboard players operation Total uhcPG /= ChunkSize uhcPG
+scoreboard players reset ChunkSize uhcPG
+
+scoreboard players operation Total uhcPG *= Total uhcPG
+
+bossbar add uhc_pack:pregen {"text":"Generating...","color":"white"}
```

Comment:
If its possible, it would be great to set the bossbar text in `stage/next` so that it can say 'Generating overworld...' / 'Generating nether...'.

Reply:
Yep, once I make that `create_bossbar` function that should be perfectly doable :)